subsys/bluetooth/mesh/access.c

```diff
@@ -116,9 +116,14 @@ static int32_t next_period(struct bt_mesh_model *mod)
 	struct bt_mesh_model_pub *pub = mod->pub;
 	uint32_t elapsed, period;
 
-	period = bt_mesh_model_pub_period_get(mod);
-	if (!period) {
-		return 0;
+	if (mod->pub->count) {
+		period = BT_MESH_PUB_TRANSMIT_INT(mod->pub->retransmit);
+		period *= BT_MESH_PUB_MSG_NUM(mod->pub);
+	} else {
+		period = bt_mesh_model_pub_period_get(mod);
+		if (!period) {
+			return 0;
+		}
 	}
 
 	elapsed = k_uptime_get_32() - pub->period_start;
```

Comment:
The warning log on line 134 will probably start showing up a lot more now, and might need more accurate text